ui/src/components/App/App.jsx

```diff
@@ -1,13 +1,77 @@
 import { useState, useEffect } from 'react'
+import * as React from "react"
 import './App.css'
 import NavBar from '../NavBar/NavBar'
 import LoggedOutView from '../LoggedOutView/LoggedOutView'
 import MessagesView from '../MessagesView/MessagesView'
 import axios from "axios"
 import {BrowserRouter,Routes,Route} from "react-router-dom"
+import * as config from '../../config'
 
-export default function App() {
-  const [isLoggedIn, setIsLoggedIn] = useState(localStorage.getItem("current_user_id") !== null)
+export default function App() { 
+  const [isLoggedIn, setIsLoggedIn]         = useState(localStorage.getItem("current_user_id") !== null)
+  const [city, setCity]                     = useState("");
+  const [state, setState]                   = useState("");
+  const [listings, setListings]              = useState([]);
+  const [properties, setProperties]         = useState([]);
+  const [average, setAverage]               = useState(0);
+  const [minPrice, setMinPrice]             = useState(0);
+  const [maxPrice, setMaxPrice]             = useState(0);
+  const [numProperties, setNumProperties]   = useState(0);
+  
+  React.useEffect(() => {
+    // Temp variables used to calculate the average rent price in a city and the
+    // number of properties in the user's price range
+     let totalPrice      = 0;
+     let totalProperties = 0;
+
+     // Iterate over all listings in the city, add the price of each listing to 
+     // calculate the average
+     listings.forEach((listing) => {
+      totalPrice += listing.price;
+
+      // If the given listing is within the user's price range, add the properties to
+      // the array of properties within the user's price range
+      // If the given listing is within the user's price range, increment the counter
+      // variable for the number of properties within the user's price range 
+      if (listing.price >= minPrice && listing.price <= maxPrice) {
+        ++totalProperties;
+        setProperties((prev) => [...prev, listing]);
+        console.log(properties);
+        console.log(listing);
+      }
+     });
+
+     totalPrice /= listings.length;
```

Comment:
check the rounding around this - and keep in mind whenever you do division, to always check if the denominator is 0 - or else your app will throw an error or show a "not enough listings found" page

Reply:
Ok I'll add rounding to two decimal places for now and the do a conditional to check if the length of the listings array is > 0.